packages/yuudachi/src/functions/logging/upsertReportLog.ts

```diff
@@ -0,0 +1,52 @@
+import type { APIEmbed, Embed, Guild, Message } from 'discord.js';
+import type { Sql } from 'postgres';
+import { container } from 'tsyringe';
+import { formatMessageToEmbed } from './formatMessageToEmbed.js';
+import { generateReportEmbed } from './generateReportEmbed.js';
+import { kSQL } from '../../tokens.js';
+import { resolveMessage } from '../../util/resolveMessage.js';
+import type { Report } from '../reports/createReport.js';
+import { checkLogChannel } from '../settings/checkLogChannel.js';
+import { getGuildSetting, SettingsKeys } from '../settings/getGuildSetting.js';
+
+export async function upsertReportLog(guild: Guild, report: Report, message?: Message) {
+	const sql = container.resolve<Sql<any>>(kSQL);
+	const reportLogChannel = checkLogChannel(guild, await getGuildSetting(guild.id, SettingsKeys.ReportChannelId));
+	const locale = await getGuildSetting(guild.id, SettingsKeys.Locale);
+
+	try {
+		if (!message && report.messageId) {
+			message = await resolveMessage(reportLogChannel!.id, report.guildId, report.channelId!, report.messageId, locale);
+		}
+	} catch {}
+
+	const author = await guild.client.users.fetch(report.authorId);
+
+	const embeds: (APIEmbed | Embed)[] = [await generateReportEmbed(author, report, locale, message)];
```

Comment:
When would this return `Embed`? We never generate anything with d.js

Reply:
Will mark this as resolved